packages/react/src/contexts/IonLifeCycleContext.tsx

```diff
@@ -63,8 +88,75 @@ export const DefaultIonLifeCycleContext = class implements IonLifeCycleContextIn
     }
   }
 
+  teardownCallback(callback: LifeCycleCallback, callbacks: any[]) {
+    // Find any destructors that have been registered for the callback
+    const matches = callbacks.filter((x) => x.id === callback.id);
+    if (matches.length !== 0) {
+      // Execute the destructor for each matching item
+      matches.forEach((match) => {
+        if (match && typeof match.destructor === 'function') {
+          match.destructor();
+        }
+      });
+      // Remove all matching items from the array
+      callbacks = callbacks.filter((x) => x.id !== callback.id);
+    } else {
+      /**
+       * If the destructor isn't registered in the array, then the
+       * user-provided callback was never invoked. This is usually
+       * the result of the useEffect hook in dev mode. So we manually
+       * invoke the callback to get the destructor to tear it down.
+       */
+      const destructor = callback();
```

Comment:
I'm questioning this now. If the `useEffect` was called and the lifecycle callback was registered, but the lifecycle event never fired, I _don't_ think we should be invoking it & then tearing it down. Likely it should never call either the lifecycle callback or the destructor in this case.

Reply:
Verified this was not needed.